task_sdk/src/airflow/sdk/execution_time/supervisor.py

```diff
@@ -0,0 +1,552 @@
+    signal.signal(signal.SIGINT, signal.SIG_DFL)
+    signal.signal(signal.SIGUSR2, signal.SIG_DFL)
+
+    if log_fd > 0:
+        # A channel that the task can send JSON-formated logs over.
+        #
+        # JSON logs sent this way will be handled nicely
+        from airflow.sdk.log import configure_logging
+
+        log_io = os.fdopen(log_fd, "wb", buffering=0)
+        configure_logging(enable_pretty_log=False, output=log_io)
+
+    last_chance_stderr = sys.__stderr__ or sys.stderr
+
+    if "PYTEST_CURRENT_TEST" in os.environ:
+        # When we are running in pytest, it's output capturing messes us up. This works around it
+        sys.stdout = sys.__stdout__
+        sys.stderr = sys.__stderr__
+
+    # Ensure that sys.stdout et al (and the underlying filehandles for C libraries etc) are connected to the
+    # pipes form the supervisor
+
+    for handle_name, sock, mode, close in (
+        ("stdin", child_stdin, "r", True),
+        ("stdout", child_stdout, "w", True),
+        ("stderr", child_stderr, "w", False),
+    ):
+        handle = getattr(sys, handle_name)
+        try:
+            fd = handle.fileno()
+            os.dup2(sock.fileno(), fd)
+            if close:
+                handle.close()
+        except io.UnsupportedOperation:
+            if "PYTEST_CURRENT_TEST" in os.environ:
+                # When we're running under pytest, the stdin is not a real filehandle with an fd, so we need
+                # to handle that differently
+                fd = sock.fileno()
+            else:
+                raise
+
+        setattr(sys, handle_name, os.fdopen(fd, mode))
+
+    def exit(n: int) -> NoReturn:
+        with suppress(ValueError, OSError):
+            sys.stdout.flush()
+        with suppress(ValueError, OSError):
+            sys.stderr.flush()
+        with suppress(ValueError, OSError):
+            last_chance_stderr.flush()
+        os._exit(n)
+
+    if hasattr(atexit, "_clear"):
+        # Since we're in a fork we want to try and clear them
+        atexit._clear()
+        base_exit = exit
+
+        def exit(n: int) -> NoReturn:
+            atexit._run_exitfuncs()
+            base_exit(n)
+
+    try:
+        target()
+        exit(0)
+    except SystemExit as e:
+        code = 1
+        if isinstance(e.code, int):
+            code = e.code
+        elif e.code:
+            print(e.code, file=sys.stderr)
+        exit(code)
+    except Exception:
+        # Last ditch log attempt
+        exc, v, tb = sys.exc_info()
+
+        import traceback
+
+        try:
+            last_chance_stderr.write("--- Last chance exception handler ---\n")
+            traceback.print_exception(exc, value=v, tb=tb, file=last_chance_stderr)
+            exit(99)
+        except Exception as e:
+            with suppress(Exception):
+                print(
+                    f"--- Last chance exception handler failed --- {repr(str(e))}\n", file=last_chance_stderr
+                )
+            exit(98)
```

Comment:
Does exit code `98` and `99` represent something? or we use it to handle the logic somewhere?

Reply:
That was the signal in BashOperator (back long time ago) to mark a task as skipped.

Reply:
Oh was it? Oops, I should change those then. These are just meant to be a marker of "hey, something _really_ odd happened in this process"

Reply:
Changed to 126 and 125, and added a comment.